java/com/google/javascript/jscomp/JsCompilerRunner.java

```diff
@@ -65,6 +68,81 @@ protected CompilerOptions createOptions() {
         options.setPropertyRenaming(PropertyRenamingPolicy.OFF);
         options.setDisambiguateProperties(false);
     }
+    if (devBuild) {
+      applySpeedyOptions(options);
+    }
     return options;
   }
+
+  /**
+   * Remove checks and undo most of the options set by the ADVANCED
+   * CompilationLevel to make development builds faster.
+   *
+   * Does not change property renaming options so code that works when compiled
+   * this way will also work with property renaming.
+   *
+   * @param options CompilerOptions object to update
+   */
+  private void applySpeedyOptions(CompilerOptions options) {
+    options.setLanguageOut(CompilerOptions.LanguageMode.ECMASCRIPT_2020);
+
+    options.resetWarningsGuard();
+
+    options.setCheckSymbols(false);
+    options.setCheckTypes(false);
+    options.setInferTypes(false);
+    options.setInferConst(false);
+    options.setCheckSuspiciousCode(false);
+    options.setRewritePolyfills(false);
+    options.setIsolatePolyfills(false);
+    options.setComputeFunctionSideEffects(false);
+
+    // All the safe optimizations.
+    options.setClosurePass(true); // Must be set to true for things to work
+    options.setFoldConstants(false);
+    options.setCoalesceVariableNames(false);
+    options.setDeadAssignmentElimination(false);
+    options.setExtractPrototypeMemberDeclarations(false);
+    options.setCollapseVariableDeclarations(false);
+    options.setConvertToDottedProperties(false);
+    options.setLabelRenaming(false);
+    options.setRemoveDeadCode(false);
```

Comment:
Dead code elimination causes problems sometimes when people don't get the entry_points right. Might be worth leaving this on too

Reply:
Done